### PR TITLE
Send extra fields from driver to solving engine

### PR DESCRIPTION
**Pull request by tokenwood**
# Description
Solving engines need to be aware of how much gas their solutions use to set the feeAmounts correctly. To do so properly, they need to be able to simulate transactions. To do so, they need to know some extra fields that were not passed on by the driver:
- router for ConstantProductPool and ConcentratedLiquidityPool
- balancer_pool_id for StablePool and WeightedProductPool

# Changes
- added missing fields in structs in auction
- set the fields when creating Auction struct 

## How to test
trivial changes, not sure if or where to add tests

### CODE REVIEW

**Comment by tokenwood** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by tokenwood** [comment]
recheck

**Review by sunce86 — CHANGES_REQUESTED** [review]
I think we miss at least two things here:
1. Update the corresponding structs on the `solvers` side.
2. Update the solvers openapi.yml

**Comment by tokenwood** [comment]
Hi @sunce86 , thanks for your comments. I ammended the commit with your suggestions.

**Comment by tokenwood** [comment]
I fixed the unit tests

File: crates/driver/src/infra/solver/dto/auction.rs [hunk 1]
```diff
@@ -304,6 +312,8 @@ struct WeightedProductPool {
     #[serde_as(as = "serde_with::DisplayFromStr")]
     id: usize,
     address: eth::H160,
+    balancer_pool_id: eth::H256,
+    vault: eth::H160,
```
Comment (fleupold) [inline crates/driver/src/infra/solver/dto/auction.rs:316]: Do you really need the vault? Isn't it always the same address in practice?

**Review by fleupold — COMMENTED** [review]
Changes look good overall to me. Not sure if we need the vault address in practice though?

**Comment by tokenwood** [comment]
@sunce86 @fleupold looks like I am breaking test-forked-node but I have no idea why and am unable to run the tests myself locally. There isn't very much doc on how to set it up, this is what I'm trying:
started the db:
`docker-compose up for the db`
started a local fork with anvil: 
`ANVIL_IP_ADDR=0.0.0.0 anvil --fork-url http://192.168.1.60:8545`
run the test:
`FORK_URL_MAINNET="http://0.0.0.0:8545" cargo test -p e2e forked_node -- --ignored`

and I get the following error:

> test limit_orders::forked_node_gnosis_single_limit_order ... FAILED
> thread 'main' panicked at /Users/runner/work/foundry/foundry/crates/anvil/src/eth/backend/mem/mod.rs:240:39:
> Failed to create genesis: GetAccount(0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266, Server returned an error response: error code -32002: No state available for block 18477910 (0x5093667b36506e36eb8a9bd602d75176be61ca080e64f4b1279bf07719211a90)

Any idea why it fails or what I'm doing wrong with the test setup? Thanks

**Comment by fleupold** [comment]
You'd need an archive node, which is secured with a secret in our CI which is not available on PRs that are merged from forked repositories. Don't worry about this failing test.

**Review by fleupold — APPROVED** [review]
LGTM, could you also upstream this change to https://github.com/gnosis/solvers

In particular https://github.com/gnosis/solvers/blob/main/src/api/routes/solve/dto/auction.rs

**Comment by tokenwood** [comment]
@fleupold like this? 
https://github.com/gnosis/solvers/pull/4

**Pull request merged**